src/top_model/main.cpp

```diff
@@ -0,0 +1,261 @@
+#include <string>
+#include <cadmium\modeling\coupling.hpp>
+#include <cadmium\modeling\ports.hpp>
+#include <cadmium\modeling\dynamic_model_translator.hpp>
+#include <cadmium\concept\coupled_model_assert.hpp>
+#include <cadmium\modeling\dynamic_coupled.hpp>
+#include <cadmium\modeling\dynamic_atomic.hpp>
+#include <cadmium\engine\pdevs_dynamic_runner.hpp>
+#include <cadmium\logger\tuple_to_ostream.hpp>
+#include <cadmium\logger\common_loggers.hpp>
+
+#include "../../include/atomics/subnet_cadmium.hpp"
+#include "..\..\lib\vendor\NDTime.hpp"
+#include "..\..\lib\vendor\iestream.hpp"
+#include "..\..\include\data_structures\message.hpp"
+#include "..\..\include\atomics\sender_cadmium.hpp"
+#include "..\..\include\atomics\receiver_cadmium.hpp"
+
+using namespace std;
+using hclock=chrono::high_resolution_clock;
+using TIME = NDTime;
+
+/***** SETING INPUT PORTS FOR COUPLEDs *****/
+struct inp_control : public cadmium::in_port<Message_t>{};
+struct inp_1 : public cadmium::in_port<Message_t>{};
+struct inp_2 : public cadmium::in_port<Message_t>{};
+
+/***** SETING OUTPUT PORTS FOR COUPLEDs *****/
+struct outp_ack : public cadmium::out_port<Message_t>{};
+struct outp_1 : public cadmium::out_port<Message_t>{};
+struct outp_2 : public cadmium::out_port<Message_t>{};
+struct outp_pack : public cadmium::out_port<Message_t>{};
+
+/********************************************/
+/****** APPLICATION GENERATOR *******************/
+/********************************************/
+
+template<typename T>
+class ApplicationGen : public iestream_input<Message_t,T> {
+	public:
+		ApplicationGen() = default;
+		ApplicationGen(const char* file_path) :
+		iestream_input<Message_t,T>(file_path) {}
+};
+
+int main(int argc, char ** argv) {
+	if (argc < 2) {
+		cout << "you are using this program with wrong parameters. "
+				"The program should be invoked as follow:";
+		cout << argv[0] << " path to the input file " << endl;
+		return 1;
+	}
+	auto start = hclock::now(); //to measure simulation execution time
+	
+	/*************** Loggers *******************/
+	static std::ofstream out_data("data/abp_output.txt");
+    struct oss_sink_provider{
+        static std::ostream& sink(){          
+            return out_data;
+        }
+    };
+    using info = cadmium::logger::logger<cadmium::logger::logger_info,
+				 cadmium::dynamic::logger::formatter<TIME>,
+				 oss_sink_provider>;
+    using debug = cadmium::logger::logger<cadmium::logger::logger_debug,
+    		      cadmium::dynamic::logger::formatter<TIME>,
+			      oss_sink_provider>;
+    using state = cadmium::logger::logger<cadmium::logger::logger_state,
+    		      cadmium::dynamic::logger::formatter<TIME>,
+			      oss_sink_provider>;
+    using log_messages = cadmium::logger::logger<cadmium::logger::logger_messages,
+    		             cadmium::dynamic::logger::formatter<TIME>,
+			             oss_sink_provider>;
+	using routing = cadmium::logger::logger<cadmium::logger::logger_message_routing,
```

Comment:
need to remove 4 spaces for indentation

Reply:
Please download the files to your local and check the indentations.
Its looking perfect in notepad++ or other editor with 4 spaces